Makefile

```diff
@@ -18,3 +18,44 @@ wasm:
 
 parser:
 	go generate .
+
+build:
+	go build -ldflags=$(LDFLAGS) ./cmd/arrai
+
+
+######## Build information report
+originalVersion=$(shell git describe --tags)
+
+ifeq (-, $(findstring -, $(originalVersion))) #it is in branch
+tagName= $(firstword $(subst -,  ,$(originalVersion)))
+diffLogs = $(foreach item, $(shell git log --pretty=format:"%h" $(tagName)..HEAD),$(item))
+
+ifeq (true, $(words $(diffLogs)) = 0 || $(shell git status -s) = ) # no changes
+Version=$(tagName)
+else
+
+ifeq (, $(shell git status -s)) # it doesn't have uncommitted or untrakced changes
+Version=$(FullCommit)~$(words $(diffLogs)) = $(tagName)
+else
+Version=DIRTY-$(FullCommit)~$(words $(diffLogs)) = $(tagName)
+endif
+
+endif
+
+else
+
+# it is in tag
+ifeq (,$(shell git status -s)) # no changes
+Version=$(versionVal)
```

Comment:
Where does this come from? Should it be originalVersion?

Reply:
Yes, it is originalVersion.